app/scripts/lib/transaction/util.ts

```diff
@@ -85,8 +85,9 @@ export async function addDappTransaction(
 
 export async function addTransaction(
   request: AddTransactionRequest,
+  internalAccount: InternalAccount[] = [],
 ): Promise<TransactionMeta> {
-  validateSecurity(request);
+  validateSecurity(request, internalAccount);
```

Comment:
Once this is also in the request, we can extract this within `validateSecurity` rather than passing down directly.